DefaultJdkTrustConfigTests fail using bundled JDK

Some tests in `DefaultJdkTrustConfigTests` fail on our bundled JDK. I presume the truststore in adoptopenjdk is slightly different than the Oracle JDK :facepalm: 

Keep in mind, to reproduce this make sure you do not have `RUNTIME_JAVA_HOME` set so that the build will default to using the bundled JDK for the test runtime JVM.

**Build scan**:
https://gradle-enterprise.elastic.co/s/tpwfhrqqwfbok/tests/:libs:elasticsearch-ssl-config:test/org.elasticsearch.common.ssl.DefaultJdkTrustConfigTests/testGetSystemTrustStoreWithNoSystemProperties#1

**Repro line**:
`./gradlew ':libs:elasticsearch-ssl-config:test' --tests "org.elasticsearch.common.ssl.DefaultJdkTrustConfigTests"`

**Reproduces locally?**:
yes

**Applicable branches**:
master, 7.x, 7.13, 7.12

**Failure excerpt**:
```
org.elasticsearch.common.ssl.DefaultJdkTrustConfigTests > testGetNonPKCS11TrustStoreWithPasswordSet FAILED
    java.lang.AssertionError: Cannot find trusted issuer with name [thawte].
        at __randomizedtesting.SeedInfo.seed([D1D1FC4A5C6CF0B4:C8E7AFA0BAF5D16E]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.common.ssl.DefaultJdkTrustConfigTests.assertHasTrustedIssuer(DefaultJdkTrustConfigTests.java:64)
        at org.elasticsearch.common.ssl.DefaultJdkTrustConfigTests.assertStandardIssuers(DefaultJdkTrustConfigTests.java:50)
        at org.elasticsearch.common.ssl.DefaultJdkTrustConfigTests.testGetNonPKCS11TrustStoreWithPasswordSet(DefaultJdkTrustConfigTests.java:40)
```

--- a/libs/ssl-config/src/test/java/org/elasticsearch/common/ssl/DefaultJdkTrustConfigTests.java
+++ b/libs/ssl-config/src/test/java/org/elasticsearch/common/ssl/DefaultJdkTrustConfigTests.java
@@ -48,8 +48,7 @@ public class DefaultJdkTrustConfigTests extends ESTestCase {
         assertHasTrustedIssuer(trustManager, "COMODO");
         assertHasTrustedIssuer(trustManager, "GlobalSign");
         assertHasTrustedIssuer(trustManager, "GoDaddy");
-        assertHasTrustedIssuer(trustManager, "certSign");
-        assertHasTrustedIssuer(trustManager, "Google Trust Services");
+        assertHasTrustedIssuer(trustManager, "QuoVadis");
         assertHasTrustedIssuer(trustManager, "Internet Security Research Group");
     }
 

--- a/libs/ssl-config/src/test/java/org/elasticsearch/common/ssl/DefaultJdkTrustConfigTests.java
+++ b/libs/ssl-config/src/test/java/org/elasticsearch/common/ssl/DefaultJdkTrustConfigTests.java
@@ -26,7 +26,6 @@ public class DefaultJdkTrustConfigTests extends ESTestCase {
 
     private static final BiFunction<String, String, String> EMPTY_SYSTEM_PROPERTIES = (key, defaultValue) -> defaultValue;
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/71717")
     public void testGetSystemTrustStoreWithNoSystemProperties() throws Exception {
         final DefaultJdkTrustConfig trustConfig = new DefaultJdkTrustConfig((key, defaultValue) -> defaultValue);
         assertThat(trustConfig.getDependentFiles(), emptyIterable());
@@ -34,7 +33,6 @@ public class DefaultJdkTrustConfigTests extends ESTestCase {
         assertStandardIssuers(trustManager);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/71717")
     public void testGetNonPKCS11TrustStoreWithPasswordSet() throws Exception {
         final DefaultJdkTrustConfig trustConfig = new DefaultJdkTrustConfig(EMPTY_SYSTEM_PROPERTIES, "fakepassword".toCharArray());
         assertThat(trustConfig.getDependentFiles(), emptyIterable());
@@ -46,11 +44,14 @@ public class DefaultJdkTrustConfigTests extends ESTestCase {
         assertThat(trustManager.getAcceptedIssuers(), not(emptyArray()));
         // This is a sample of the CAs that we expect on every JRE.
         // We can safely change this list if the JRE's issuer list changes, but we want to assert something useful.
-        assertHasTrustedIssuer(trustManager, "VeriSign");
-        assertHasTrustedIssuer(trustManager, "GeoTrust");
         assertHasTrustedIssuer(trustManager, "DigiCert");
-        assertHasTrustedIssuer(trustManager, "thawte");
         assertHasTrustedIssuer(trustManager, "COMODO");
+        assertHasTrustedIssuer(trustManager, "GlobalSign");
+        assertHasTrustedIssuer(trustManager, "GoDaddy");
+        assertHasTrustedIssuer(trustManager, "Amazon");
+        assertHasTrustedIssuer(trustManager, "certSign");
+        assertHasTrustedIssuer(trustManager, "Google Trust Services");
+        assertHasTrustedIssuer(trustManager, "Internet Security Research Group");
     }
 
     private void assertHasTrustedIssuer(X509ExtendedTrustManager trustManager, String name) {

--- a/libs/ssl-config/src/test/java/org/elasticsearch/common/ssl/DefaultJdkTrustConfigTests.java
+++ b/libs/ssl-config/src/test/java/org/elasticsearch/common/ssl/DefaultJdkTrustConfigTests.java
@@ -48,7 +48,6 @@ public class DefaultJdkTrustConfigTests extends ESTestCase {
         assertHasTrustedIssuer(trustManager, "COMODO");
         assertHasTrustedIssuer(trustManager, "GlobalSign");
         assertHasTrustedIssuer(trustManager, "GoDaddy");
-        assertHasTrustedIssuer(trustManager, "Amazon");
         assertHasTrustedIssuer(trustManager, "certSign");
         assertHasTrustedIssuer(trustManager, "Google Trust Services");
         assertHasTrustedIssuer(trustManager, "Internet Security Research Group");

